hash/_wasm/src/lib.rs

```diff
@@ -1,52 +1,262 @@
 // Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
 
-use digest::{Digest, DynDigest};
+use digest::{Digest, DynDigest, ExtendableOutput, Reset, Update, VariableOutput, XofReader};
 use wasm_bindgen::prelude::*;
 
+/// A Hasher for any supported cryptographic hash algorithm, wrapping them in
+/// a common interface for our WASM use.
 #[wasm_bindgen]
-pub struct DenoHash {
-  inner: Box<dyn DynDigest>,
+#[derive(Clone)]
+pub struct Hasher {
+    algorithm: &'static str,
+    inner: HashBox,
 }
```

Comment:
Can you run `cargo fmt`? In particular, deno uses two spaces for indentation, not four.

Reply:
Copied in deno core's `.rustfmt.toml` and applied it.